packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -439,22 +438,28 @@ export class FluidDataStoreRuntime
 			id = idArg;
 			this.validateChannelId(id);
 		} else {
+			/**
+			 * There is currently a bug where certain data store ids such as "[" are getting converted to ASCII characters
+			 * in the snapshot. So, return uuid() here for now while we identify and fix the issue.
+			 */
+			id = uuid();
+
 			// We use three non-overlapping namespaces:
 			// - detached state: even numbers
 			// - attached state: odd numbers
 			// - uuids
 			// In first two cases we will encode result as strings in more compact form, with leading underscore,
 			// to ensure no overlap with user-provided DDS names (see validateChannelId())
-			if (this.visibilityState !== VisibilityState.GloballyVisible) {
-				// container is detached, only one client observes content, no way to hit collisions with other clients.
-				id = encodeCompactIdToString(2 * this.contexts.size, "_");
-			} else {
-				// Due to back-compat, we could not depend yet on generateDocumentUniqueId() being there.
-				// We can remove the need to leverage uuid() as fall-back in couple releases.
-				const res =
-					this.dataStoreContext.containerRuntime.generateDocumentUniqueId?.() ?? uuid();
-				id = typeof res === "number" ? encodeCompactIdToString(2 * res + 1, "_") : res;
-			}
+			// if (this.visibilityState !== VisibilityState.GloballyVisible) {
```

Comment:
Should we put these changes behind a feature flag, for easy toggles in DF/PROD?

Reply:
I am not sure I see the value in doing that because we can't enable it unless we fix the bug which will require making code changes anyway. We can enable it via that change and add feature flags if needed.

Reply:
I hear you there. Feature flag might come handy, if the issue happens to be on ODSP side, and we can just toggle our change off without making a code change. Since the true-root-cause is still being figured out, there may be value in keeping this change behind a switch.

Reply:
I can add it behing a feature. But I doubt we will use it (Loop would probably not let us it being so risky). Also, we heard from ODSP that it's not their issue most likely.
Let me work on adding a feature, it doesn't hurt :).

Reply:
Added feature flags